target.go

```diff
@@ -163,6 +163,27 @@ func (s *RecursiveResolver) parseQueryFromRequest(r *http.Request) (*dns.Msg, er
 	}
 }
 
+func isSignedByParent(domainName string, rrs []dns.RR) bool {
+	segments := strings.Split(domainName, ".")
+	numberSegments := len(segments)
+	// fail fast
+	if numberSegments < 0 {
+		return false
+	}
+	parentDomain := strings.Join(segments[1:], ".")
+	for _, rrRecord := range rrs {
+		switch rrRecord.(type) {
+		case *dns.RRSIG:
+			t := (rrRecord).(*dns.RRSIG)
```

Comment:
You can actually include this in the type switch. Saves a line of code, and `t` will always have the correct type if you add more `case`s.

```suggestion
		switch t := rrRecord.(type) {
		case *dns.RRSIG:
```